src/hooks.js

```diff
@@ -135,3 +135,62 @@ export function useRepoDetails(baseUrl, detailsUrl) {
   }
   return usePromise(fetchDescription, [detailsUrl], null)
 }
+
+export function useEsc(cb, deps) {
+  const handlekeyDown = useCallback(
+    e => {
+      if (e.keyCode === keycodes.esc) {
+        cb()
+      }
+    },
+    [cb]
+  )
+  useEffect(() => {
+    window.addEventListener('keydown', handlekeyDown)
+    return () => window.removeEventListener('keydown', handlekeyDown)
+  }, [handlekeyDown])
+}
+
+const QUERY_VAR = '?labels='
+// checks if query string var exists
+// parses data and validates data consistency (will throw if prop don't exist)
+export function useSharedLabels(dao) {
+  const [isSharedLink, setIsSharedLink] = useState(false)
+  const [sharedLabels, setSharedLabels] = useState([])
+
+  const removeSharedLink = useCallback(
+    () => (window.location.hash = `#/${dao}`),
+    [dao]
+  )
+
+  useEffect(() => {
+    const index = window.location.hash.indexOf(QUERY_VAR)
+    if (index > -1) {
+      const raw = window.location.hash.substr(index + QUERY_VAR.length)
+      try {
+        const data = JSON.parse(window.decodeURI(atob(raw)))
+        setSharedLabels(data.map(({ address, name }) => ({ address, name })))
+        setIsSharedLink(true)
+      } catch (e) {
+        console.warn(
+          'There was an error parsing/validating the shared data: ',
+          e
+        )
+      }
+    }
+  }, [])
+
+  return { isSharedLink, setIsSharedLink, sharedLabels, removeSharedLink }
+}
+
+export function useSelected(initial) {
+  const [selected, setSelected] = useState(initial)
+  const [allSelected, someSelected] = useMemo(
+    () => [
+      Array.from(selected.values()).every(Boolean),
+      Array.from(selected.values()).some(Boolean),
+    ],
+    [selected]
+  )
+  return { selected, setSelected, allSelected, someSelected }
```

Comment:
This also feels like we could include some of the toggling functionality (all and individual ids) to avoid repeating it.

Should we also use this directly in `SelectableLocalIdentity`?